DOCUMENTATION.md

```diff
@@ -3927,11 +3931,178 @@ const cfg = {
 The client calls the `init()` method with the `properties` configuration option. Then the client calls the `getSSLOptions()`
 method of `SSLOptionsFactory` to create the `options` object.
 
-## 9.2. Credentials
+## 9.2 Authentication
 
-One of the key elements in Hazelcast security is the `Credentials` object, which can be used to carry all security attributes of
-the Hazelcast Node.js client to Hazelcast members. Then, Hazelcast members can authenticate the clients and perform access
-control checks on the client operations using this `Credentials` object.
+By default, the client does not use any authentication method and just uses the configured cluster name to connect to members.
+
+Hazelcast Node.js client has more ways to authenticate the client against the members, using the ``security`` configuration.
+
+The ``security`` configuration has an element called ``credentials`` in which you can specify different kinds of authentication mechanisms.
+
+The ``credentials`` must always have a property called ``type`` that describes the valid types of credentials,
+which is one of the case-insensitive versions below.
+
+- ``username_password``
+- ``token``
+- ``custom``
+
+The content of the ``credentials`` depends on the ``type`` used, and is described below in detail.
+The idea behind specifying the ``type`` in the ``credentials`` is to prevent possible misuses of that configuration
+element and fail-fast on invalid configurations.
+
+### 9.2.1. Username Password Authentication
+
+The client can authenticate with username and password against the members with the following configuration.
+The ``type`` of the ``credentials`` is case-insensitive ``username_password`` string, and the properties
+are ``username`` and ``password`` strings.
+
+```js
+const config = {
+    security: {
+        credentials: {
+            type: 'username_password',
+            username: 'admin',
+            password: 'some-strong-password'
+        }
```

Comment:
Does it make sense to move type up, like `security` would contain one of `username_password`, `token` or `custom`? That would remove a level and makes the configuration a bit easier:
```
const config = {
    security: {
        username_password: {
            username: 'admin',
            password: 'some-strong-password'
        }
```

Reply:
Makes sense, I have simplified the configuration as suggested.

Reply:
FWIW the .NET configuration file (JSON) supports writing the following "shortcuts", (with the `"authentication"` block being a top-level configuration item):
```json
"authentication": {
    "username-password": {
        "username": "<username>",
        "password": "<password>"
    }
}
```
Or:
```json
"authentication": {
    "token": {
        "encoding": "<none>|<base64",
        "data": "..."
    }
}
```
Or:
```json
"authentication": {
    "kerberos": {
        "spn": "..."
    }
}
```
In addition to the generic one:
```json
"authentication": {
    "credentialsFactory": {
        "typeName": "My.Project.CredentialsFactory, My.Project",
        "args": {
            "arg1": "<arg>",
            "arg2": "123456"
        }
    }
}
```

Reply:
The final form is similar to what we have in C#